sched/task/task_delete.c

```diff
@@ -143,6 +144,15 @@ int task_delete(pid_t pid)
       goto errout;
     }
 
+#ifdef HAVE_GROUP_MEMBERS
+  /* Kill all of the children of the task.  This will not kill the currently
+   * running task/pthread (this_task).  It will kill the main thread of the
+   * task group if this_task is a pthread.
+   */
+
+  group_kill_children(rtcb);
```

Comment:
why is this necessary?
why is this rtcb, not dtcb?

Reply:
> why is this necessary?

I think that It's dangerous to terminate the main thread but leave other threads in the task keep running. And other similar function also do the same thing(e.g. task_restart, nxsig_abnormal_termination, exit...).

> why is this rtcb, not dtcb?
good catch, I will fix it.

Reply:
why is it dangerous? it's something i believe the most of pthread implementations allow.